static/solo/js/libs/player.js

```diff
@@ -74,6 +74,12 @@
       overlayColor: isset(options.overlay_color) ? options.overlay_color : null,
       overlayOpacity: isset(options.overlay_opacity) ? options.overlay_opacity : null
     });
+
```

Comment:
You can do something like

```
var advertisingOptions = isset(options.advertising) && "object" === typeof options.advertising && !isEmpty(options.advertising) ? options.advertising : {};
this.advertising = compact({
	enabled: isset(advertisingOptions.enabled) ? advertisingOptions.enabled : null
	adServerUrl: isset(advertisingOptions.adServerUrl) ? advertisingOptions.adServerUrl : null,
	...
});
```

Then in the player settings you do like:

`advertising: that.advertising`